src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -538,15 +545,20 @@ export class IndexPatternsService {
     }
 
     const body = indexPattern.getAsSavedObjectBody();
-    const response = await this.savedObjectsClient.create(savedObjectType, body, {
-      id: indexPattern.id,
-    });
-    indexPattern.id = response.id;
-    this.indexPatternCache.set(indexPattern.id, Promise.resolve(indexPattern));
+    const response: SavedObject<IndexPatternAttributes> = (await this.savedObjectsClient.create(
+      savedObjectType,
+      body,
+      {
+        id: indexPattern.id,
+      }
+    )) as SavedObject<IndexPatternAttributes>;
+
+    const createdIndexPattern = await this.initFromSavedObject(response);
+    this.indexPatternCache.set(createdIndexPattern.id!, Promise.resolve(createdIndexPattern));
     if (this.savedObjectsCache) {
       this.savedObjectsCache.push(response as SavedObject<IndexPatternSavedObjectAttrs>);
     }
-    return indexPattern;
+    return createdIndexPattern;
```

Comment:
This is the main change. I want returned index pattern to go through the all transformations that are happening when saved objects is saved